### DISCUSSION TITLE
Dynamic directory registration of headless server

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#2888] [Ideas]
The gui mode of the server can change from public to private and vice versa without having to restart the server itself.
Can there be added a means to do same in headless mode, perhaps sending a different kill signal, similar to enable/disable recording?

**Comment by hoffie** [comment]
I thought we already had something for that in the JSON-RPC API, but it seems like this detail can only be read and not changed right now: https://github.com/jamulussoftware/jamulus/blob/master/docs/JSON-RPC.md#jamulusservergetserverprofile

I think that would be the place to add such logic though. The number of UNIX signals is limited (and we already use USR1 and USR2, I think) and is -- as the name says -- limited to *nix-style systems.

**Reply by pljones** [reply]
Yes, it would be through the JSONRPC interface.  It should be pretty easy to add.
